src/bsoncxx/document/view.hpp

```diff
@@ -29,27 +29,97 @@ namespace bsoncxx {
 BSONCXX_INLINE_NAMESPACE_BEGIN
 namespace document {
 
+///
+/// A read-only, non-owning view of a BSON document.
+///
 class BSONCXX_API view {
 
    public:
     class iterator;
     class const_iterator;
 
+    ///
+    /// @returns A const_iterator to the first element of the document.
+    ///
     const_iterator cbegin() const;
+
+    ///
+    /// @returns A const_iterator to the past-the-end element of the document.
+    ///
     const_iterator cend() const;
 
+    ///
+    /// @returns An iterator to the first element of the document.
+    ///
     iterator begin() const;
+
+    ///
+    /// @returns An iterator to the past-the-end element of the document.
+    ///
     iterator end() const;
 
     // TODO switch to stdx::string_view
+
+    ///
+    /// Finds the first element of the document with the provided key. If there is
+    /// no such element, the past-the-end iterator will be returned. The runtime of
+    /// find() is linear in the length of the document.
+    ///
+    /// @remark In BSON, keys are not required to be unique. If there are multiple
+    /// elements with a matching key in the document, the first matching element from
+    /// the start will be returned.
+    ///
+    /// @param key
+    ///   The key to search for.
+    ///
+    /// @return An iterator to the matching element, if found, or the past-the-end iterator.
+    ///
     iterator find(const string_or_literal& key) const;
+
+    ///
+    /// Finds the first element of the document with the provided key. If there is no
+    /// such element, the invalid element will be returned. The runtime of operator[]
+    /// is linear in the length of the document.
```

Comment:
Does it search recursively or only the top level? Do you think it would be important to mention this or is it obvious?